recipes/srt/all/conanfile.py

```diff
@@ -1,103 +1,118 @@
             self.requires("pthreads4w/3.0.0")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                              "set (CMAKE_MODULE_PATH \"${CMAKE_CURRENT_SOURCE_DIR}/scripts\")",
-                              "list(APPEND CMAKE_MODULE_PATH \"${CMAKE_CURRENT_SOURCE_DIR}/scripts\")")
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["ENABLE_APPS"] = False
-        self._cmake.definitions["ENABLE_LOGGING"] = False
-        self._cmake.definitions["ENABLE_SHARED"] = self.options.shared
-        self._cmake.definitions["ENABLE_STATIC"] = not self.options.shared
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_APPS"] = False
+        tc.variables["ENABLE_LOGGING"] = False
+        tc.variables["ENABLE_SHARED"] = self.options.shared
+        tc.variables["ENABLE_STATIC"] = not self.options.shared
         if self._has_stdcxx_sync:
-            self._cmake.definitions["ENABLE_STDCXX_SYNC"] = True
-        self._cmake.definitions["ENABLE_ENCRYPTION"] = True
-        self._cmake.definitions["USE_OPENSSL_PC"] = False
-        if self.settings.compiler == "Visual Studio":
+            tc.variables["ENABLE_STDCXX_SYNC"] = True
+        tc.variables["ENABLE_ENCRYPTION"] = True
+        tc.variables["USE_OPENSSL_PC"] = False
+        if is_msvc(self):
             # required to avoid warnings when srt shared, even if openssl shared,
             # otherwise upstream CMakeLists would add /DELAYLOAD:libeay32.dll to link flags
-            self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+            tc.variables["OPENSSL_USE_STATIC_LIBS"] = True
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
+            'set (CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts")',
+            'list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/scripts")',
+        )
+        if not self._has_posix_threads and not self._has_stdcxx_sync:
+            save(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                "find_package(pthreads4w REQUIRED)\n"
+                "link_libraried(pthreads4w::pthreads4w)\n",
+                append=True)
```

Comment:
there is a typo: `link_libraried`

andI don't think it can work anyway, because link_libraries() only affect targets created after, but you append to top CMakeLists, so it will be no-op.

Reply:
My bad. Thanks!